setup/install_gatling/30_classification/90_clamps.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+# Clamps is a module based stress testing tool. It will generate a number of
+# users on a system to simulate agents. It will then randomize their puppet
+# agent -t times to simulate a large number of agents.
+def add_clamps_group_ca(classifier, pe_infra)
+  # The Clamps::master class manages auth.conf for the agents - by setting it
+  # to allow all.  This is insecure and worth looking at overriding at some
+  # point, but this should only be used for metrics / load testing on a private
+  # network, so shouldn't be an issue short term.
+  clamps_ca_group = {
+    'name'    => 'Clamps CA',
+    'rule'    => ['or', ['=', 'name', master.node_name]], # pinned node
+    'parent'  => pe_infra['id'],
+    'classes' => {
+      'clamps::master' => {}
+    }
+  }
+  classifier.find_or_create_node_group_model(clamps_ca_group)
+end
+
+# Creates a node group for creating the fake agents. This node group should
+# match all users that are running as root.
+def add_clamps_group_agent(classifier, pe_infra)
+  clamps_agents_group = {
+    'name'    => 'Clamps - Agent Nodes',
+    'rule'    => ['and', ['=', %w[fact id], 'root'], ['=', 'name', metric.node_name]],
+    'parent'  => pe_infra['id'],
+    'classes' => {
+      'clamps::agent' => {
+        'amqpass'               => '',
+        'master'                => any_hosts_as?(:loadbalancer) ? 'puppet' : master.node_name,
+        'nonroot_users'         => options[:scale][:num_nonroot_users]       || 2,
+        'daemonize'             => options[:scale][:daemonize]               || false,
+        'mco_daemon'            => options[:scale][:mco_daemon],
```

Comment:
where does the value come from?  I don't see it set anywhere.

Reply:
This would be set in the beaker options. I have included it for backward compatibility. If it is unset, the value is removed from the parameter list by the `reject` call at the end of the hash.  For current and future PE versions this value would not be set since we are no longer using MCO.